src/servicesHost.ts

```diff
@@ -96,8 +96,7 @@ export function makeServicesHost(
 
     getProjectReferences: () => projectReferences,
 
-    getScriptFileNames: () =>
-      [...files.keys()].filter(filePath => filePath.match(scriptRegex)),
+    getScriptFileNames: () => Array.from(instance.rootFileNames),
```

Comment:
This was the change I mentioned before that fixes https://github.com/TypeStrong/ts-loader/issues/949. Doing this exposed a couple other problems, like needing to add the file to `instance.rootFileNames` when `onlyCompileBundledFiles` is true.